lib/child-writer/writer-proxy.js

```diff
@@ -109,62 +122,77 @@ permissions.isElevated().then((elevated) => {
   console.log('Re-spawning with elevation')
 
   return new Bluebird((resolve, reject) => {
-    ipc.config.id = process.env.IPC_CLIENT_ID
-    ipc.config.silent = true
 
-    // > If set to 0, the client will NOT try to reconnect.
-    // See https://github.com/RIAEvangelist/node-ipc/
-    //
-    // The purpose behind this change is for this process
-    // to emit a "disconnect" event as soon as the GUI
-    // process is closed, so we can kill the CLI as well.
-    ipc.config.stopRetrying = 0
+    let child = null
+
+    /**
+     * @summary Emit an object message to the IPC server
+     * @function
+     * @private
+     *
+     * @param {Buffer} data - json message data
+     *
+     * @example
+     * emitMessage(Buffer.from(JSON.stringify({
+     *   foo: 'bar'
+     * })));
+     */
+    const emitMessage = (data) => {
+      // Output from stdout/stderr coming from the CLI might be buffered,
+      // causing several progress lines to come up at once as single message.
+      // Trying to parse multiple JSON objects separated by new lines will
+      // of course make the parser confused, causing errors later on.
+      _.each(utils.splitObjectLines(data.toString()), (object) => {
+        ipc.of[process.env.IPC_SERVER_ID].emit('message', object)
```

Comment:
As `ipc.of[process.env.IPC_SERVER_ID].emit('message', object)` is also used in `onError` below, would it be worth abstracting that part out?

Reply:
I don't think 2 repetitions of a single (partial) line, which is just a method call warrant any abstraction